mono/metadata/threads.c

```diff
@@ -5079,26 +5131,25 @@ mono_threads_join_threads (void)
 void
 mono_thread_join (gpointer tid)
 {
-#ifndef HOST_WIN32
-	pthread_t thread;
 	gboolean found = FALSE;
+	gpointer value;
 
 	joinable_threads_lock ();
 	if (!joinable_threads)
 		joinable_threads = g_hash_table_new (NULL, NULL);
-	if (g_hash_table_lookup (joinable_threads, tid)) {
+
+	value = g_hash_table_lookup (joinable_threads, tid);
```

Comment:
Please use `g_hash_table_lookup_extended` so a `NULL` value wouldn't be considered false.

Reply:
will do